test/rules/parameter_assignments.dart

```diff
@@ -45,3 +49,34 @@ void otherBad1(int parameter) { // LINT
   parameter += 3;
```

Comment:
I think you know what I think :-)

The principle I would apply here is that errors should be reported as close as possible to the location where the problem occurs. This is a corollary to the more general principle that tools should do work so that people don't have to.

The problem isn't at the point where the parameter is declared, it's at the point where the value of the parameter is changed. Hence, I think we should change the location of the lint error.